--- a/src/components/RecipientBubble.vue
+++ b/src/components/RecipientBubble.vue
@@ -76,7 +76,7 @@
 						:taggable="true"
 						track-by="label"
 						:multiple="false"
-						:placeholder="t('name', 'Contact name …')"
+						:placeholder="t('mail', 'Contact name …')"
 						:clear-search-on-select="true"
 						:show-no-options="false"
 						:append-to-body="false"
--- a/src/components/TranslationModal.vue
+++ b/src/components/TranslationModal.vue
@@ -30,7 +30,7 @@
 					class="translate-dialog__select"
 					input-id="to"
 					label="name"
-					:aria-label-combobox="t('spreed', 'Target language to translate into')"
+					:aria-label-combobox="t('mail', 'Target language to translate into')"
 					:placeholder="t('mail', 'Translate to')"
 					:options="availableOutputLanguages"
 					no-wrap />
--- a/src/service/NotificationService.js
+++ b/src/service/NotificationService.js
@@ -77,7 +77,7 @@
 export function showNewMessagesNotification(messages) {
 	showNotification(
 		t('mail', 'Nextcloud Mail'),
-		getNotificationBody(messages),
+		t('mail', 'You have a new secure message in ITSL Hubs'),
 		generateFilePath('mail', 'img', 'mail-notification.png'),
 	)
 }
